xarray/tests/test_backends.py

```diff
@@ -1561,10 +1575,14 @@ def test_weakrefs(self):
 class TestPyNioAutocloseTrue(TestPyNio):
     autoclose = True
 
+    @requires_pynio
+    def setUp(self):
+        pass
+
```

Comment:
This doesn't seem to be working. I'll try to come up with something else but if anyone has any ideas for how to get around this, I'm all ears.

Reply:
Does it work if you hand-write the test-skipping decorator to raise `unittest.SkipTest` rather than relying on `pytest.mark.skipif`?